### PR TITLE
Create filter grid markup

### PR DESCRIPTION
**Pull request by castelgo**
First component version

### CODE REVIEW

File: src/pods/movies/components/movies-filter.component.tsx [hunk 1]
```diff
@@ -1,4 +1,134 @@
...
+  genres: GenreEntity[];
+}
+
+export class MoviesFilterComponentInner extends React.Component<Props, State> {
+
+  state: State = {
+    movieFilter: {
+      title: '',
+      genre: '',
+      year: 2018,
```
Comment (brauliodiez) [inline src/pods/movies/components/movies-filter.component.tsx]: That's a good case... why 2018? :)

Movie filter (ViewModel) should have year field as string, to handle the empty case (blank string) but convert to number for the api calls (for the state it should be string as well to hold the empty string).

File: src/pods/movies/components/movies-filter.component.tsx [hunk 2]
```diff
@@ -1,4 +1,134 @@
...
+    })
+  }
+
+  updateTitle = (title: string) =>
+    this.setState({
+      movieFilter: {
+        ...this.state.movieFilter,
+        title
+      }
+    });
```
Comment (brauliodiez) [inline src/pods/movies/components/movies-filter.component.tsx]: We should go for a generic approach (in the future we could have more filters), not sure if this applies to combos

```typescript
  updateTitle = (fieldId: string) => (value: any) =>
    this.setState({
      movieFilter: {
        ...this.state.movieFilter,
        [fieldId]: value,
      }
    });
```

File: src/pods/movies/components/movies-filter.component.tsx [hunk 3]
```diff
@@ -1,4 +1,134 @@
...
+            variant="contained"
+            color="primary"
+            className={classes.button}
+            onClick={() => this.props.onApplyFilter(this.state.movieFilter)}
+          >
+            Apply filter
+          </Button>
+
+        </div>
+      </>
```
Comment (brauliodiez) [inline src/pods/movies/components/movies-filter.component.tsx]: This is a bit hard to read I would think about how to componentize this.

File: src/pods/movies/components/movies-filter.component.tsx (hunk 1 above)
Comment (chalsy87) [inline src/pods/movies/components/movies-filter.component.tsx]: are we sure that we want this value by default? first loading doesn't apply any filter, and we are implementing to type filters, first apply all filters, but the second one applies only the filter which has changed.

File: src/pods/movies/reducer/filter.spec.ts [hunk 4]
```diff
@@ -14,15 +14,15 @@ describe('moveies/reducer/filterReducer tests', () => {
...
+        expect(nextState.genre).toEqual('');
         expect(nextState.year).toBeNull();
     });
 
     it('should return same state without mutate it when passing state and some action type', () => {
         // Arrange
         const state: MovieFilter = {
             title: 'test title',
-            genere: 'test genere',
+            genre: 'test genere',
```
Comment (brauliodiez) [inline src/pods/movies/reducer/filter.spec.ts]: typo on the literal 
'test genre'

File: src/pods/movies/view-model.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export interface MovieFilter {
+    title: string;
+    genre: string;
+    year: number;
```
Comment (brauliodiez) [inline src/pods/movies/view-model.ts]: year should be string

File: src/scenes/scene-movies-list.component.tsx [hunk 6]
```diff
@@ -2,8 +2,10 @@ import * as React from 'react';
 import { FlexView, SingleView } from '../layout';
 import { MoviesFilterContainer, MoviesGridContainer } from '../pods/movies';
 
+const applyFilter = () => null
+
 export const SceneMovieList: React.StatelessComponent = () => 
  <FlexView>
-    <MoviesFilterContainer />
+    <MoviesFilterContainer onApplyFilter={applyFilter}/>
```
Comment (brauliodiez) [inline src/scenes/scene-movies-list.component.tsx]: onApply filter should not be on the container expose to the outer world

**Comment by castelgo** [comment]
Pull request updated. All changes done.

**Pull request merged**